File: ResourceManager/Models/Configurations/UserRoleConfiguration.vb
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
Imports Microsoft.EntityFrameworkCore
Imports ResourceManager.Configuration

Namespace ResourceManager.Models.Configurations
    Public Class UserRoleConfiguration
        Implements IEntityTypeConfiguration(Of UserRole)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of UserRole)) Implements IEntityTypeConfiguration(Of UserRole).Configure
            builder.ToTable("UserRoles")
            builder.HasKey(Function(x) New With {x.UserID, x.RoleID
            })

            builder.HasOne(Function(x) x.User).WithMany(Function(x) x.UserRoles).HasForeignKey(Function(x) x.UserID)

            builder.HasOne(Function(x) x.Role).WithMany(Function(x) x.UserRoles).HasForeignKey(Function(x) x.RoleID)

            builder.HasData(DesignData.Instance.UserRoles)
        End Sub
    End Class
End Namespace

File: ResourceManager/Models/Configurations/UserRoleConfiguration.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System.Collections;
using System.Collections.Generic;
using ResourceManager.Configuration;

namespace ResourceManager.Models.Configurations
{
    public class UserRoleConfiguration : IEntityTypeConfiguration<UserRole>
    {
        public void Configure(EntityTypeBuilder<UserRole> builder)
        {
            builder.ToTable("UserRoles");
            builder.HasKey(x => new { x.UserID, x.RoleID });

            builder.HasOne(x => x.User)
                .WithMany(x => x.UserRoles)
                .HasForeignKey(x => x.UserID);

            builder.HasOne(x => x.Role)
                .WithMany(x => x.UserRoles)
                .HasForeignKey(x => x.RoleID);

            builder.HasData(DesignData.Instance.UserRoles);
        }
    }
}
